### PR TITLE
added more mui classes for buttons

### PR DESCRIPTION
**Pull request by JeevaRamanathan**
Address #588
Added `size` and `variant` for the button, along with testcase and example. Please let me know if this is on the right track and required any changes.

### CODE REVIEW

File: taipy/gui/_renderers/factory.py [hunk 1]
```diff
@@ -99,6 +99,8 @@ class _Factory:
                 ("active", PropertyType.dynamic_boolean, True),
                 ("hover_text", PropertyType.dynamic_string),
                 ("width", PropertyType.string_or_number),
+                ("size", PropertyType.dynamic_string),
```
Comment (FredLL-Avaiga) [inline taipy/gui/_renderers/factory.py]: no need for these to be dynamic (string is enough)

File: frontend/taipy-gui/src/components/Taipy/Button.tsx [hunk 2]
```diff
@@ -40,7 +44,8 @@ const Button = (props: ButtonProps) => {
     const className = useClassNames(props.libClassName, props.dynamicClassName, props.className);
     const active = useDynamicProperty(props.active, props.defaultActive, true);
     const hover = useDynamicProperty(props.hoverText, props.defaultHoverText, undefined);
-
+    const variant = useDynamicProperty(props.variant,props.defaultVariant,"outlined") as "outlined"| "text" | "contained";
+    const size = useDynamicProperty(props.size,props.defaultSize,"medium") as "small"| "medium" | "large";
```
Comment (FredLL-Avaiga) [inline frontend/taipy-gui/src/components/Taipy/Button.tsx]: no need to make these dynamic

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
the viselements.json file should be also impacted.

File: frontend/taipy-gui/src/components/Taipy/Button.tsx (hunk 2 above)
Comment (JeevaRamanathan) [reply frontend/taipy-gui/src/components/Taipy/Button.tsx]: Hi @FredLL-Avaiga, I couldn’t find any hooks for string similar to useDynamicProperty. Does this approach is valid here?
`const variant = (props.variant ?? "outlined") as "outlined" | "text" | "contained"`

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Button.tsx]: No need for hooks
You can simply set the default value when you destructure the props

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
Tx

File: frontend/taipy-gui/src/components/Taipy/Button.tsx (hunk 2 above)
Comment (JeevaRamanathan) [reply frontend/taipy-gui/src/components/Taipy/Button.tsx]: @FredLL-Avaiga Quick question: a bit confused, is the updated code (doesn't use hooks) I used fine, or does it need any changes?

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Button.tsx]: Please change it
You can set the default value in the line destructuring the props
Something like
const { variant = "outlined", ... } = props;

Comment (JeevaRamanathan) [reply frontend/taipy-gui/src/components/Taipy/Button.tsx]: Done, I’ve changed it. Thanks!

**Comment by JeevaRamanathan** [comment]
Hi @FredLL-Avaiga, the changes have been made. Please have a look based on your time.

**Comment by FredLL-Avaiga** [comment]
We're discussing with @FabienLelaquais what should be done ...

**Comment by JeevaRamanathan** [comment]
@quest-bot loot #588

**Review by FabienLelaquais — APPROVED** [review]
Good job

**Review by FabienLelaquais — APPROVED** [review]
👍

**Review by FredLL-Avaiga — APPROVED** [review]

**Pull request merged**